tests/components/myuplink/test_config_flow.py

```diff
@@ -158,3 +158,161 @@ async def test_flow_reauth(
     assert len(hass.config_entries.async_entries(DOMAIN)) == 1
     assert len(mock_setup.mock_calls) == 1
     assert mock_config_entry.data["token"]["scope"] == CURRENT_SCOPE
+
+
+@pytest.mark.usefixtures("current_request_with_host")
+async def test_flow_reauth_account_mismatch(
```

Comment:
Can we maybe paremetrize this test and `test_flow_reauth_bad_scope` by making the scope a parameter? That would remove duplication, WDYT?

Reply:
OK, collapsing to one reauth flow test instead of three.

Reply:
Keep in mind that we don't want if statements in tests

Reply:
Ah yes this works beautifully